### PR TITLE
Air gap

### PR DESCRIPTION
**Pull request by JakeMathews**
Allows users to specify a path to the inspector they want to use. 
Gradle requires a path to a libs folder with all the necessary jar files. 
Nuget requires a path to the inspector's .nupkg file. User's can also now specify the repo they want the inspector to use when resolving dependencies. NOTE: This will require IntegrationNugetInspector version 2.0.0 which is not yet released. So this shouldn't be merged until such time.

This also includes improvements to the build file such as pulling down these required files at build time so they can be distributed to customers as well as a new way of sending information about the build for detect to use at runtime.

### CODE REVIEW

**Comment by JakeMathews** [comment]
The build will not succeed until version 2.0.0 of IntegrationNugetInspector is released.

File: src/main/resources/version.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+1.1.0-SNAPSHOT
```
Comment (jrichardBD) [inline src/main/resources/version.txt]: Why is this getting checked in? Wasn't this being generated by the build? And don't your changes make this file obsolete?

**Comment by jrichardBD** [comment]
Why are we replacing the version text? It seems that the buildInfo approach is more code and more complex

File: src/main/resources/version.txt (hunk 1 above)
Comment (JakeMathews) [reply src/main/resources/version.txt]: Yes this shouldn't be there. It must have accidentally go thrown in there when switching between branches.

**Comment by JakeMathews** [comment]
We are replacing version text because the initial implementation of air-gap required additional info from the build to be sent to hub-detect. After talking with Eric, we agreed that it would be useful to leave this in, in the event that in the future, we need to send any other info from the build.

**Pull request merged**